src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -305,47 +307,52 @@ internal TAttribute ResolveFromBindings(IReadOnlyDictionary<string, object> bind
 
-        // Resolution policy for { } in  binding templates. 
-        // The default policy is just a direct substitution for the binding data. 
-        // Derived policies can enforce formatting / escaping when they do injection. 
-        private class ResolutionPolicy
+        internal IResolutionPolicy<TAttribute> GetPolicy(PropertyInfo propInfo)
         {
-            public string TemplateBind(BindingTemplate template, IReadOnlyDictionary<string, object> bindingData)
+            AutoResolveAttribute autoResolveAttribute = propInfo.GetCustomAttribute<AutoResolveAttribute>();
+
+            // Special-case Table as there is no way to declare this ResolutionPolicy
+            // and use BindingTemplate in the Core assembly
+            var formatterType = autoResolveAttribute.ResolutionPolicyType;
+
+            if (formatterType != null)
             {
-                if (bindingData == null)
+                if (formatterType == typeof(ODataFilterResolutionPolicyMarker<TAttribute>))
+                {
+                    return new ODataFilterResolutionPolicy<TAttribute>();
+                }
+
+                if (!typeof(IResolutionPolicy<TAttribute>).IsAssignableFrom(formatterType))
                 {
-                    return template.Pattern;
+                    throw new InvalidOperationException($"Formatter Policy on {propInfo.Name} must derive from {typeof(IResolutionPolicy<TableAttribute>).Name} and can't be of type {formatterType.Name}.");
                 }
-                return this.TemplateBindWorker(template, bindingData);
+
+                var obj = Activator.CreateInstance(formatterType);
+                return (IResolutionPolicy<TAttribute>)obj;
             }
 
-            protected virtual string TemplateBindWorker(BindingTemplate template, IReadOnlyDictionary<string, object> bindingData)
-            {                
-                return template.Bind(bindingData);
+            // return the default policy
+            if (autoResolveAttribute.ResolutionPolicyType == null)
+            {
+                return _defaultResolutionPolicy;
             }
 
-            // Expand on [AutoResolve] property to make this pluggable, 
-            public static ResolutionPolicy GetPolicy(PropertyInfo propInfo)
+            IResolutionPolicy<TAttribute> policy = null;
+
+            if (autoResolveAttribute != null && autoResolveAttribute.ResolutionPolicyType != null)
             {
-                if (propInfo.DeclaringType == typeof(TableAttribute) && propInfo.Name == "Filter")
+                var ctor = autoResolveAttribute.ResolutionPolicyType.GetConstructor(new Type[] { typeof(IResolutionPolicy<TAttribute>) });
+                if (ctor != null)
                 {
-                    return new TableFilterResolutionPolicy();
+                    policy = (IResolutionPolicy<TAttribute>)Activator.CreateInstance(autoResolveAttribute.ResolutionPolicyType, new object[] { _defaultResolutionPolicy });
```

Comment:
I think I'm going to remove this "nesting" feature to keep the change simpler. None of us need it right now and we can always expand it in the future.